src/main.cpp

```diff
@@ -3056,9 +3074,16 @@ bool SendMessages(CNode *pto)
             }
         }
 
-        // Request the next blocks. Mostly this will get exucuted during IBD but sometimes even
-        // when the chain is syncd a block will get request via this method.
-        requester.RequestNextBlocksToDownload(pto);
+        {
+            TRY_LOCK(cs_main, locked);
+            static int rarely = 0;
+            if (locked && ((++rarely) & 7) == 0)
+            { // I don't need to deal w/ blocks as often as tx and this is time consuming
```

Comment:
Confirmed, this does have quite a negative effect on IBD.  I can understand this area is probably a performance hog but for the time being, since we have so little time, can I suggest the following (which I've tested IBD on) and then we can revisit this area when you come back.

-TRY_LOCK(cs_main, locked);
-static int rarely = 0;
-if (locked && ((++rarely) & 7) == 0)
+LOCK(cs_main);

Reply:
let me try just the try_lock first.  What happens is that all 4 net processing threads get locked up waiting for cs_main.  If its locked, we might as well just have 1 of them in there and three of them doing other things.  

The tests consistently pass on my machine, but actually I've just been thinking that if the number of connections and the number of threads is a multiple of 8, the & 7 logic might consistently ignore calling the function for some nodes, until some event jiggles the order, whereupon it consistently ignores some other nodes.